### ISSUE TITLE
Authoring API: `index` is 0-based while `action_number`/`condition_number` are 1-based; a wrong index silently rewrites another row

### ISSUE THREAD

**Issue by PaulSnow** [DTRules/DTRules#1297]
## What happens

`dtrules table patch` mixes two numbering conventions in one payload shape:

| Field | Ops | Base |
|---|---|---|
| `action_number` | `update-action`, `update-action-dsl`, `delete-action` | **1-based** (matches `<action_number>` in the XML) |
| `condition_number` | `update-condition`, `update-condition-dsl`, `delete-condition` | **1-based** |
| `index` | `update-initial-action`, `delete-initial-action`, `update-context`, `delete-context` | **0-based** |

Nothing in the payload says which is which, and `table schema --patch` documents `index` only as `{"type": "integer", "minimum": 0}`.

## Why it bites

A patch with a 1-based `index` **succeeds** and silently rewrites the *next* row:

```
$ echo '{"op":"update-initial-action","index":7,"dsl":"set strBuffer = current time"}' \
    | dtrules table patch Run_Test_10 --project sampleprojects/SyntaxTests
{ "status": "patched", ... }
```

The intended row (1-based 7) was untouched; initial action 8, `set street = using client (street)`, was destroyed. The result is a valid project that verifies and builds — the damage is only visible by diffing the DSL.

This happened while working #1280 and cost a rule; it was caught by diffing, not by any check.

`index: 0` is also ambiguous: with `omitempty` it cannot be distinguished from "field not supplied", so `{"op":"delete-initial-action"}` with no index deletes the first initial action.

## Fix

Make `index` 1-based, like the other two, and reject `0`/missing with an error that says so. The stored XML is unaffected — this is an input convention only — but it is a breaking change for any caller currently passing 0-based indexes.

## Affected

`cmd/dtrules/table_patch.go` (the four ops above), the `--patch` schema text, `dtrules docs` authoring section, and any MCP write path that shares the payload.
